sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java

```diff
@@ -836,11 +837,16 @@ public void run() {
                         Map.Entry<UUID, ErrorLogReport> unprocessedEntry = unprocessedIterator.next();
                         ErrorLogReport errorLogReport = unprocessedEntry.getValue();
                         if (errorLogReport.report.getThrowable() instanceof NativeException) {
-                            Exception exception = errorLogReport.log.getException();
-                            dumpFile = new File(exception.getStackTrace());
-                            exception.setStackTrace(null);
-                            byte[] logfileContents = FileManager.readBytes(dumpFile);
-                            dumpAttachment = ErrorAttachmentLog.attachmentWithBinary(logfileContents, "minidump.dmp", "application/octet-stream");
+                            String minidumpFilePath = errorLogReport.log.getException().getMinidumpFilePath();
+
+                            /* It can be null when NativeException is incorrectly used or there is already stored invalid data. */
+                            if (minidumpFilePath != null) {
+                                dumpFile = new File(minidumpFilePath);
+                                byte[] logfileContents = FileManager.readBytes(dumpFile);
+                                dumpAttachment = ErrorAttachmentLog.attachmentWithBinary(logfileContents, "minidump.dmp", "application/octet-stream");
+                            }
+                            errorLogReport.log.getException().setMinidumpFilePath(null);
+                            errorLogReport.log.getException().setStackTrace(null);
```

Comment:
This line is not needed, we are using minidump field now.

Reply:
I think it's good to have both since that's an implementation detail and if that changed in the future we could regress

Reply:
It's dead code, stack trace will always be null here.